mantisshrimp/models/mantis_rcnn/mantis_faster_rcnn.py

```diff
@@ -4,16 +4,54 @@
 from mantisshrimp.core import *
 from mantisshrimp.models.mantis_rcnn.rcnn_param_groups import *
 from mantisshrimp.models.mantis_rcnn.mantis_rcnn import *
+from mantisshrimp.backbones import *
 
 
 class MantisFasterRCNN(MantisRCNN):
     @delegates(FasterRCNN.__init__)
-    def __init__(self, n_class, h=256, pretrained=True, metrics=None, **kwargs):
+    def __init__(
+        self,
+        n_class,
+        h=256,
+        backbone="default",
+        pretrained=True,
+        metrics=None,
+        **kwargs,
+    ):
         super().__init__(metrics=metrics)
-        self.n_class, self.h, self.pretrained = n_class, h, pretrained
-        self.m = fasterrcnn_resnet50_fpn(pretrained=self.pretrained, **kwargs)
-        in_features = self.m.roi_heads.box_predictor.cls_score.in_features
-        self.m.roi_heads.box_predictor = FastRCNNPredictor(in_features, self.n_class)
+        self.n_class, self.h, self.backbone, self.pretrained = (
+            n_class,
+            h,
+            backbone,
+            pretrained,
+        )
+
+        if self.backbone == "defualt":
+            # Creates the default fasterrcnn as given in pytorch. Trained on COCO dataset
+            self.m = fasterrcnn_resnet50_fpn(pretrained=self.pretrained, **kwargs)
+            in_features = self.m.roi_heads.box_predictor.cls_score.in_features
+            self.m.roi_heads.box_predictor = FastRCNNPredictor(
+                in_features, self.n_class
+            )
+        else:
+            # Creates the custom backbone model trained on ImageNet.
+            try:
+                self.base_model = create_torchvision_backbone(
+                    backbone=self.backbone, is_pretrained=pretrained
+                )
+                self.ft_anchor_generator = AnchorGenerator(
+                    sizes=((32, 64, 128)), aspect_ratios=((0.5, 1.0, 2.0))
+                )
+                self.ft_mean = [0.485, 0.456, 0.406]  # imagenet mean and std
+                self.ft_std = [0.229, 0.224, 0.225]
+                self.m = FasterRCNN(
+                    backbone=self.base_model,
+                    num_classes=self.n_class,
+                    image_mean=self.ft_mean,
+                    image_std=self.ft_std,
```

Comment:
We should pass kwargs to `FasterRCNN` here, so the user can still specify parameters if we wants to

Reply:
This is gonna be done.